### ISSUE TITLE
StartExecution() Issue?

### ISSUE THREAD

**Issue by jaypman** [gurmitteotia/guflow#45]
Thanks for providing this library and extensive documentation.  This is giving me a good start in SWF :)

I tried  running your guflow-samples (ChildWorkflow) however I noticed an issue that I cannot figure out.  
I cannot trigger an execution for the workflow via code.  

For example:
workflowHost.StartExecution();
activityHost.StartExecution();

I see the Workflow Types registered and also the Activity Types registered in the domain successfully. 

However, I do not see see any executions started in the SWF Dashboard (Workflow Executions).

The only way I can execute the workflows is manually via the AWS Dashboard.  

I cannot seem to programmatically trigger it via .StartExecution() method.

Please advise..

**Comment by gurmitteotia** [comment]
Hi,

You can start the workflow as documented here- https://github.com/gurmitteotia/guflow/wiki/Start-Workflow

For more code examples you can look at integration tests. I'm just listing one of the test here-

https://github.com/gurmitteotia/guflow/blob/master/Guflow.IntegrationTests/ActivityInputTests.cs

**Comment by jaypman** [comment]
Thank you for the quick response :)  I apologize if the answer is already in the documentation already.  However I'm still struggling to execute a workflow.  Maybe I'm not understanding the concepts correctly.

I'm trying follow the examples using "**host**" to execute but having no success.  Is the recommend way to execute workflow is to use **StartWorkflowRequest.For** instead of **domain.StartWorkflow()**  or **host.StartExecution()** ?

Here is the code snippet from [guflow-samples](https://github.com/gurmitteotia/guflow-samples)/[ChildWorkflow](https://github.com/gurmitteotia/guflow-samples/tree/master/ChildWorkflow)/Program.cs
           
            var client = new AmazonSimpleWorkflowClient(new BasicAWSCredentials("xxxx", "xxxx"), RegionEndpoint.USEast1);
            var domain = new Domain("GuflowTestDomain", client);
            await domain.RegisterAsync(10, "test guflow");
            await domain.RegisterWorkflowAsync<ParentWorkflow>();
            await domain.RegisterWorkflowAsync<KidPlayWorkflow>();

            var activities = new[] { typeof(PlayOnSwing), typeof(PlayOnZipWire) };
            foreach (var activity in activities)
            {
                await domain.RegisterActivityAsync(activity);
            }
            using (var workflowHost = domain.Host(new Workflow[] { new ParentWorkflow(), new KidPlayWorkflow() }))
            using (var activityHost = domain.Host(activities))
            {
                workflowHost.OnError(e =>
                {
                    Console.WriteLine(e.Exception); return ErrorAction.Continue;
                });
              
                workflowHost.StartExecution();  // Not executing (no errors)
                activityHost.StartExecution();  // Not executing (no errors)
                Console.WriteLine("Press a key to terminate");
                Console.ReadKey();
            }

I also modified the code to use

`var request = StartWorkflowRequest.For<ParentWorkflow>("4")`

however it did not execute the workflow according to the SWF Dashboard.

![image](https://user-images.githubusercontent.com/26774764/158941346-e871cf69-f00f-4996-aa18-f397c3a23bb4.png)


I do see the Workflow Types and Activity Types registered in the Dashboard

**Comment by gurmitteotia** [comment]
Hi,

The purpose of WorkflowHost.StartExecution is to put the WorkflowHost in state where it will execute the hosted workflows when they started/triggered in Amazon SWF. When WorkflowHost is executing, it will poll the Amazon SWF for events and interpret them against the hosted workflows as documented here- https://github.com/gurmitteotia/guflow/wiki/Error-handling#workflow-host-exceptions


The purpose of Domain. StartWorkflowAsync is to start the workflow in Amazon SWF.  You can start/trigger the workflow either form AWS Console, as you're already doing, or using Domain.StartWorkflowAsync API. You'll likely to use the API approach because you'll want to start the workflow based on some logic.  I don't know the layout of your code but Domain.StartWorkflowAsync will likely be called from different project then where workflow host is executing.

You'll need to use both APIs and they are not exclusive.

In other words starting of workflows and execution of workflows are two different things. 

To explore it further here is what you can do-
- Create a separate console application which when started will call Domain.StartWorkflowAsync for ParentWorkflow
- Keep guflow sample as it is and start its execution
- Once guflow sample is running trigger the workflow by starting the project you created in step 1

**Comment by jaypman** [comment]
Fantastic! :) I finally got is working based on your suggestions.  Thanks again for this great library! :)
